File: FE/GymTonic-FE/src/app/Models/searchedUser.ts


export class SearchedUser {

  id:number;
  username:string;
  name:string;
  lastname:string;
  exerciseIds:number[];
  workoutIds:number[];
  planIds:number[];

  constructor(id:number, username:string, name:string,
    lastname:string, exerciseIds:number[], workoutIds:number[],
    plansIds:number[]){
      this.id = id;
      this.username = username;
      this.name = name;
      this.lastname = lastname;
      this.exerciseIds = exerciseIds;
      this.workoutIds = workoutIds;
      this.planIds = plansIds;
    }
}

File: GTUser/src/main/java/com/DavideDalSanto/GTUser/Models/SearchedUser.java
package com.DavideDalSanto.GTUser.Models;

import lombok.*;

import javax.persistence.ElementCollection;
import java.util.ArrayList;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SearchedUser {
    private Long id;
    private String username;
    private String name;
    private String lastname;

    @Builder.Default
    private List<Long> userExercisesId= new ArrayList<>();
    @Builder.Default
    private List<Long> userWorkoutsId = new ArrayList<>();
    @Builder.Default
    private List<Long> userPlansIds= new ArrayList<>();


}
